platforms/core-runtime/build-operations-trace/src/main/java/org/gradle/internal/operations/trace/BuildOperationTrace.java

```diff
@@ -498,6 +498,14 @@ private static JsonGenerator createJsonGenerator() {
         return new JsonGenerator.Options()
             .addConverter(new JsonClassConverter())
             .addConverter(new JsonThrowableConverter())
+            /*
+              From ResolutionFailureData, we want to exclude the actual failure field
+              from JSON serialization in the trace, because it can contain arbitrary
+              data including circular references that aren't serializable.  See the
+              note on AmbiguousArtifactTransformsFailure for more.  Once that class
+              is refactored, we should explore removing this exclusion.
+            */
+            .excludeFieldsByName("resolutionFailure")
```

Comment:
The options are to ignore fields by name or by type.  Even Converters are only added by type.  I tried ignoring by class with a `Class.forName`, but that fails due to several  `error: package com.sun.tools.javac.util is not visible` errors and I didn't want to start exporting new packages to this project's build.

Fortunately this field name is unique in Gradle.  This is also only temporary, until we can create some new lighterweight classes to represent the artifact transform process in failures.

Reply:
I vaguely recall that BuildOperationTrace used Groovy serialization and the serialization format could be customized via annotations. I cannot seem to find that code now. I wanted to suggest to use that feature, maybe it's gone now. Otherwise I'm fine with the exclusion.

Reply:
Annotations were the first thing I looked for.  The Groovy `DefaultJsonGenerator` doesn't support them, unfortunately.  We have to (and should) migrate this to use Jackson first.

Reply:
Implementing `CustomOperationTraceSerialization` is also an option, but the problematic types aren't operations, and I don't want to define a custom format for all problem operations.